Remake the built-in color schemes to modern standards

romainl opened the pull request:
- replace the original color schemes with the remakes
- improve check_colors.vim
- rewrite check_colors.vim in vim9script
- update CODEOWNERS
- update `test_gui.vim` to match new `torte.vim`

Commit:
Remake the built-in color schemes to modern standards

- replace the original color schemes with the remakes
- improve check_colors.vim
- rewrite check_colors.vim in vim9script
- update CODEOWNERS

k-takata commented on .github/CODEOWNERS:
```diff
@@ -19,6 +19,26 @@ runtime/autoload/tar.vim		@cecamp
+runtime/colors/ron.vim			@habamax @romainl @neutaaaaan
+runtime/colors/shine.vim		@habamax @romainl @neutaaaaan
+runtime/colors/slate.vim		@habamax @romainl @neutaaaaan
+runtime/colors/tools			@habamax @romainl @neutaaaaan
```
Maybe one of line 22 or line 39 can be removed.

romainl replied:
Fixed in #[236442a](https://github.com/vim/vim/pull/9795/commits/236442aeac91f56326ff5164713dd2569ccc792c).

Commit:
Remove redundant entry from CODEOWNERS

Commit:
Fix test_gui.vim

bfrg commented on runtime/colors/tools/check_colors.vim:
```diff
@@ -1,238 +1,209 @@
 
-let s:save_cpo= &cpo
+var save_cpo = &cpo
 set cpo&vim
```
You don't need to save, set and restore `cpoptions` in a `vim9script` file. See the third paragraph under `:h E1101`.

romainl replied:
Fixed, thank you.

bfrg commented on runtime/colors/tools/check_colors.vim:
```diff
@@ -1,238 +1,209 @@
-  call cursor(1,1)
-  let err = {}
+def Test_check_colors()
+    var savedview = winsaveview()
```
Variables that don't change can be declared constant. Vim supported this even before `vim9script`.

Writing
```vim
const savedview = winsaveview()
```
shows the clear intention of the variable. Plus it catches errors early. It protects you from accidentally changing variables that aren't intended to be changed. In some languages it's called _const-correctness_.

Same can be done for the variables `hi_groups`, `bg_set`, `bg_let`, `bg_pat`, `line`  below.

habamax replied:
I have fixed it in vim/colorschemes repo, I think @romainl will add it to PR.

romainl replied:
Fixed. Could you check?

Commit:
Fix &cpo issue in check_colors.vim

Commit:
Fixed several issues pointed out in vim/colorschemes

- spell-related highlight groups
- message-related highlight groups
- FoldColumn
- ColorColumn

Commit:
Fix Windows console-related issues

This fixes a number of issues related to the way the Windows console
handles stylistic attributes like underline, bold, etc.

Commit:
Fix DiffText legibility and TabLineSel bug

Commit:
New highlight groups and small fixes

- added CurSearch
- added CursorLineFold
- added CursorLineSign
- fixed Spell* issues
- updated colors/README.txt

romainl committed:
Merge branch 'master' into feat/colorscheme-remakes
